src/Hal.c

```diff
@@ -132,89 +151,82 @@ void extract_mck_frequency()
 
 	extract_main_oscilator_frequency();
 	apply_plla_config(&master_clock_config);
-	
-	switch (master_clock_config.presc)
-	{
-		case Pmc_MasterckPresc_1:
-		{
-			break;
-		}
-		case Pmc_MasterckPresc_2:
-		{
-			mck_frequency = mck_frequency / 2;
-			break;
-		}
-		case Pmc_MasterckPresc_4:
-		{
-			mck_frequency = mck_frequency / 4;
-			break;
-		}
-		case Pmc_MasterckPresc_8:
-		{
-			mck_frequency = mck_frequency / 8;
-			break;
-		}
-		case Pmc_MasterckPresc_16:
-		{
-			mck_frequency = mck_frequency / 16;
-			break;
-		}
-		case Pmc_MasterckPresc_32:
-		{
-			mck_frequency = mck_frequency / 32;
-			break;
-		}
-		case Pmc_MasterckPresc_64:
-		{
-			mck_frequency = mck_frequency / 64;
-			break;
-		}
+
+	switch (master_clock_config.presc) {
+	case Pmc_MasterckPresc_1: {
+		break;
+	}
+	case Pmc_MasterckPresc_2: {
+		mck_frequency = mck_frequency / 2;
+		break;
+	}
+	case Pmc_MasterckPresc_4: {
+		mck_frequency = mck_frequency / 4;
+		break;
+	}
+	case Pmc_MasterckPresc_8: {
+		mck_frequency = mck_frequency / 8;
+		break;
+	}
+	case Pmc_MasterckPresc_16: {
+		mck_frequency = mck_frequency / 16;
+		break;
+	}
+	case Pmc_MasterckPresc_32: {
+		mck_frequency = mck_frequency / 32;
+		break;
+	}
+	case Pmc_MasterckPresc_64: {
+		mck_frequency = mck_frequency / 64;
+		break;
+	}
 #if defined(N7S_TARGET_SAMV71Q21)
-		case Pmc_MasterckPresc_3:
-		{
-			mck_frequency = mck_frequency / 7;
-			break;
-		}
+	case Pmc_MasterckPresc_3: {
+		mck_frequency = mck_frequency / 7;
+		break;
+	}
 #endif
 	}
 
-	switch (master_clock_config.divider)
-	{
-		case Pmc_MasterckDiv_1:
-		{
-			break;
-		}
-		case Pmc_MasterckDiv_2:
-		{
-			mck_frequency = mck_frequency / 2;
-			break;
-		}
+	switch (master_clock_config.divider) {
+	case Pmc_MasterckDiv_1: {
+		break;
+	}
+	case Pmc_MasterckDiv_2: {
+		mck_frequency = mck_frequency / 2;
+		break;
+	}
 	}
 }
 
 bool Hal_Init(void)
 {
 	reloads_counter = 0u;
-    
-  	Pmc_init(&pmc, Pmc_getDeviceRegisterStartAddress());
-  	Pmc_enablePeripheralClk(&pmc, Pmc_PeripheralId_Tc0Ch0);
 
-	extract_mck_frequency();
+	Init_setup_watchdog();
+	Pmc_init(&pmc, Pmc_getDeviceRegisterStartAddress());
+	Pmc_enablePeripheralClk(&pmc, Pmc_PeripheralId_Tc0Ch0);
 
-  	Nvic_setInterruptHandlerAddress(Nvic_Irq_Timer0_Channel0, timer_irq_handler);
-	Nvic_enableInterrupt(Nvic_Irq_Timer0_Channel0);
+	// nvic cannot be used for registration of interrupt handlers:
+	// instead, the rtems api shall be used
+	// the interrupt vector table is managed by rtems
+	rtems_interrupt_handler_install(23, "timer0", RTEMS_INTERRUPT_UNIQUE,
```

Comment:
Do we have a named value for 23? Avoid magic numbers. I can allow "4" in case for "Pmc_MasterckPresc_4", but (23, ...) is not readable at all.